graphReconstruct/graphReconstruct.go

```diff
@@ -2,230 +2,89 @@ package graphReconstruct
-	recon := fasta.Read(reconFilename)
-
-	answer := make(map[string]float64)
-
-	for i := 0; i < len(sim); i++ {
-		mistakes = 0.0
-		found = false
-		for j := 0; j < len(recon); j++ {
-			if sim[i].Name == recon[j].Name {
-				found = true
-				//DEBUG: log.Printf("\n%s \n%s \n", dna.BasesToString(sim[i].Seq), dna.BasesToString(recon[j].Seq))
-				for k := 0; k < len(sim[0].Seq); k++ {
-					if sim[i].Seq[k] != recon[j].Seq[k] {
-						mistakes = mistakes + 1
-					}
-				}
-			}
-		}
-		if found == false {
-			log.Fatal("Did not find all simulated sequences in reconstructed fasta.")
-		}
-		accuracy := mistakes / float64(len(sim[i].Seq)) * 100.0
-		//DEBUG: fmt.Printf("tot: %f, len(sim): %f, len(sim[0].Seq): %f \n", tot, float64(len(sim)), float64(len(sim[0].Seq)))
-		acc := 100 - accuracy
-		answer[sim[i].Name] = acc
-		total = total + mistakes
-	}
-	accuracy := total / (float64(len(sim)) * float64(len(sim[0].Seq))) * 100.0
-	//DEBUG: fmt.Printf("tot: %f, len(sim): %f, len(sim[0].Seq): %f \n", tot, float64(len(sim)), float64(len(sim[0].Seq)))
-	acc := 100 - accuracy
-	answer[allNodes] = acc
-	return answer
-}
-
-//write assigned sequences at all nodes to a fasta file
-func WriteTreeToFasta(tree *expandedTree.ETree, outFile string) {
-	var fastas []*fasta.Fasta
-	nodes := expandedTree.GetTree(tree)
-
-	for i := 0; i < len(nodes); i++ {
-		fastas = append(fastas, nodes[i].Fasta)
-	}
-	fasta.Write(outFile, fastas)
+type graphColumn struct {
+	AlignId    int
+	AlignNodes map[string][]*simpleGraph.Node
 }
 
-//write assigned sequences at leaf nodes to a fasta file
-func WriteLeavesToFasta(tree *expandedTree.ETree, leafFile string) {
-	var leafFastas []*fasta.Fasta
-	nodes := expandedTree.GetLeaves(tree)
-
-	for i := 0; i < len(nodes); i++ {
-		leafFastas = append(leafFastas, nodes[i].Fasta)
-	}
-	fasta.Write(leafFile, leafFastas)
-}
-
-//calculate probability of switching from one base to another
-func Prob(a int, b int, t float64) float64 {
-	var p float64
-	switch {
-	case a > 3 || b > 3:
-		p = 0
-	case a == b:
-		p = 1 - t
-	default:
-		p = t / 3
-	}
-	return p
-}
-
-//take in probability of all 4 bases return integer value of the most likely base
-func Yhat(r []float64) int {
-	var n float64
-	n = 0
-	var pos int
-	for p, v := range r {
-		if v > n {
-			n = v
-			pos = p
+//BuildNodes uses a graphColumn to create nodes for an ancestor's graph seq that represents all the unique sequences in an aligned graph
+func BuildNodes(root *expandedTree.ETree, column graphColumn, id uint32) uint32 {
+	var nodeInfo = make(map[string]bool)
+	for _, nodes := range column.AlignNodes { //nodes is all nodes for an individual species
+		for n := range nodes { //n is an individual node of an individual species
+			stringSeq := dna.BasesToString(nodes[n].Seq)
+			nodeInfo[stringSeq] = true
 		}
 	}
-	return pos
-}
-
-func allZero(r []float64) bool {
-	for _, v := range r {
-		if v != 0 {
-			return false
-		}
+	for seq := range nodeInfo {
+		id += 1
```

Comment:
I think I would increment id at the end of the loop instead of the beginning.  I think it fits style a little better to use the id you are given and pass back the id that the next function can begin using